2016-10-30  Tanzeela Naimah Khan  <tanzeela@lnxsrv05.seas.ucla.edu>
            * timespec.h: fixed integer overflow

--- lib/timespec.h	2016-11-01 20:47:20.515341000 -0700
+++ lib/timespec.h	2016-11-01 20:49:50.159114000 -0700
@@ -45,8 +45,13 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  long long int diff = (long long int) a.tv_sec - (long long int) b.tv_sec;
+  if (diff == 0)
+    return 0;
+  else if (diff > 0)
+    return 1;
+  else if (diff < 0)
+    return -1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
